py/modbuiltins.c

```diff
@@ -460,8 +460,33 @@ MP_DEFINE_CONST_FUN_OBJ_1(mp_builtin_repr_obj, mp_builtin_repr);
 STATIC mp_obj_t mp_builtin_round(size_t n_args, const mp_obj_t *args) {
     mp_obj_t o_in = args[0];
     if (MP_OBJ_IS_INT(o_in)) {
-        return o_in;
-    }
+        if(n_args == 1 || mp_obj_is_true(mp_binary_op(MP_BINARY_OP_MORE_EQUAL, args[1], mp_obj_new_int(0)))){
+            return o_in;
+        }
+        if (!MP_OBJ_IS_INT(args[1])){
+            mp_raise_TypeError("Second parameter must be an integer");
+        }
+        mp_obj_t num_dig = mp_unary_op(MP_UNARY_OP_NEGATIVE, (args[1]));
+        mp_obj_t mult = mp_binary_op(MP_BINARY_OP_POWER, mp_obj_new_int(10), num_dig);
```

Comment:
You can use `MP_OBJ_NEW_SMALL_INT(10)` to make it simpler and more efficient.  Then also `MP_OBJ_NEW_SMALL_INT(-num_dig)`.